_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: post
+title: "Technical Debt: The First 50 years"
+date: 2020-07-01
+tags: debt devops dora 
+author: Matt Tyler
+image: 
+---
+
+'Technical debt' is a term used by engineers to describe a large number of problems that may be happening during development and operation of software. It can be difficult to get support to address technical debt due to a luck of trust between engineering functions and the wider business, and the failure of engineering teams to measure and communicate technical debt in specific terms. Technical Debt itself can't be measured but as most engineers will agree that it has an impact on the ability to both deliver and operate software, the four-key metrics presented in the DORA report can act as a suitable alternative. It is easier to argue for investment in specific areas if they can be shown to have an impact on a measure that the business understands to be linked to performance and therefore the profitability of the business.
```

Comment:
>due to a luck of trust

due to a ~luck~ lack of trust

Reply:
> Technical Debt itself can't be measured

Technical Debt can be measured, but it's not easy.  

There are a number of metric that provide information on technical debt:
cyclometric complexity, coupling measures, impact on estimates.

However, the above metrics are difficult to sell in isolation - they don't talk of business impact.

The 4 key metrics give you a direct correlation between technical debt and the delivery/availability effectiveness.

Reply:
I'm not feeling the hook - why should I continue reading?

Reply:
There's no units though; I can't say I have "5 technical debts". Likewise, debt is understanding - We could write everything in haskell and the cyclometric complexity would hit zero, but that would move the debt elsewhere - everyone would have to learn haskell.

I think technical debt is heavily contextual.